src/NuGet.Core/NuGet.Build.Tasks/StaticGraphRestoreArguments.cs

```diff
@@ -0,0 +1,206 @@
+                        arguments.Options = ReadDictionary(reader);
+                    }
+                    else if (string.Equals(propertyName, nameof(EntryProjectFilePath), StringComparison.Ordinal))
+                    {
+                        arguments.EntryProjectFilePath = ReadValue(reader);
+                    }
+                }
+            }
+
+            return arguments;
+
+            Dictionary<string, string> ReadDictionary(JsonReader reader)
+            {
+                if (!reader.Read() || reader.TokenType != JsonToken.StartObject)
+                {
+                    return null;
+                }
+
+                Dictionary<string, string> dictionary = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+                while (reader.Read() && reader.TokenType == JsonToken.PropertyName && reader.Value is string key && reader.Read() && reader.TokenType == JsonToken.String && reader.Value is string value)
+                {
+                    dictionary[key] = value;
+                }
+
+                return dictionary;
+            }
+
+            string ReadValue(JsonReader reader)
+            {
+                if (reader.Read() && reader.TokenType == JsonToken.String && reader.Value is string value)
+                {
+                    return value;
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Writes the current arguments to the specified stream.
+        /// </summary>
+        /// <param name="stream">The <see cref="Stream" /> to write the arguments to.</param>
+        internal void Write(FileStream stream)
+        {
+            using (var streamWriter = new StreamWriter(stream))
+            using (var writer = new JsonTextWriter(streamWriter))
+            {
+                writer.Formatting = Formatting.Indented;
+
+                writer.WriteStartObject();
+                {
+                    WriteProperty(writer, nameof(EntryProjectFilePath), EntryProjectFilePath);
+                    WriteProperty(writer, nameof(MSBuildExeFilePath), MSBuildExeFilePath);
+                    WriteDictionary(writer, nameof(GlobalProperties), GlobalProperties);
+                    WriteDictionary(writer, nameof(Options), Options);
+                }
+                writer.WriteEndObject();
+            }
+
+            void WriteDictionary(JsonTextWriter writer, string propertyName, Dictionary<string, string> dictionary)
+            {
+                writer.WritePropertyName(propertyName);
+
+                writer.WriteStartObject();
+                foreach (KeyValuePair<string, string> option in dictionary)
+                {
+                    writer.WritePropertyName(option.Key, escape: true);
+                    writer.WriteValue(option.Value);
+                }
+                writer.WriteEndObject();
+            }
+
+            void WriteProperty(JsonTextWriter writer, string name, string value)
```

Comment:
There's some JsonUtility methods that might be helpful in reducing the code changes. 
There's a few generic methods we use for writing the other files, such as `WriteObject`. 
You don't have to change it, but sometihng to consider.

Reply:
I did see some methods in JsonUtility but they are currently non-generic:

https://github.com/NuGet/NuGet.Client/blob/a56c42dbdd901c0c611eece400ce8565fffdfdf7/src/NuGet.Core/NuGet.ProjectModel/JsonUtility.cs#L74-L78

I'm fine with adding these methods in JsonUtility but then then `NuGet.ProjectModel` would need to be loaded which may or may not already be happening.  Just tell me what you prefer :)

Reply:
There's some generic methods `<T>`. I'd rather you duplicate stuff if you can't use the existing methods.

Reply:
Okay for now I'll leave the custom logic.